vic/drivers/shared_image/src/vic_restore.c

```diff
@@ -945,9 +947,13 @@ check_init_state_file(void)
             log_err("Error reading data from \"%s\" in %s",
                     global_domain.info.lat_var, filenames.init_state);
         }
-        for (i = 0; i < global_domain.n_ny; i++) {
-            if (!assert_close_double(dvar[i],
-                                     global_domain.locations[i].latitude, rtol,
+        // implicitly nested loop over ni and nj with i set to 0;
+        // j stride = n_nx
+        for (j = 0; j < global_domain.n_ny; j++) {
+            if (!assert_close_double(dvar[j],
+                                     global_domain.locations[j *
+                                                             global_domain.n_nx]
+                                     .latitude, rtol,
```

Comment:
This is for the 1-d lat/lon case - as determined by the grid domain file, not the state file.  Accordingly, if it's the 1-d case, then all lats will match the furthest west ones (in the grid from the grid domain file, parameter file, and forcings, since they are all consistent, right?).  Meanwhile, again because its the 1-d case, there is only 1 dimension to check in the state file's lat var - one lat value per grid row.  We already checked earlier whether the dimensionality of the lat and lon variables was what it should be.  So this check should be sufficient.  At least I think so.